### PR TITLE
Pop up

### LINKED ISSUE #8: [4pt]  - Create feature: add new comment - Student B
Implement the "add new comment" form on the Comments pop up according to the wireframe:

Screenshot 2021-06-27 at 22 32 36

When the user clicks on the "Comment" button, the data is recorded in the [Involvement API](https://www.notion.so/microverse/Involvement-API-869e60b5ad104603aa6db59e08150270) and the screen is updated.

### LINKED ISSUE #11: [3pt]  - Display comments for a given item on the Comments pop-up - Student B
When the popup loads, the webapp retrieves data from the [Involvement API](https://www.notion.so/microverse/Involvement-API-869e60b5ad104603aa6db59e08150270) to show the item comments.

Screenshot 2021-06-27 at 22 21 00

This task **does not** include the counter of the comments.

### LINKED ISSUE #14: [4pt]  -Display Comments pop up with selected item's details - Student B
Create a comments popup only with the top section (displaying details of the selected item)

Screenshot 2021-06-27 at 22 10 45

When the popup loads, the webapp retrieves data from the selected API and shows details about the selected item.

Add event to button prepared by your teammate that will open your popup. You need to communicate with your team member to make sure that you do not block each other (most likely the easiest solution is to merge the list of items feature to the dev branch before you try to add that event).

### PR DESCRIPTION
**Pull request by benwmx**
* Add popup section.
* Add basic styling to the popup.
* Add Comment class for adding & getting comments from the involvement API.
please review this pull request @ahzia  and give your feedback.

### CODE REVIEW

File: src/JS/Comment.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      body: JSON.stringify({
+        item_id: id,
+        username: name,
+        comment,
+      }),
+    });
+  }
+
+  async getList(id) {
+    const response = await fetch(`${this.url}?item_id=${id}`);
```
Comment (ahzia) [inline src/JS/Comment.js]: use `.then` in order to make sure the API request returned the data successfully.
``await fetch(`${this.url}?item_id=${id}`).then((response) => response.json()).then((data) => {this.list = data});``

File: src/index.html [hunk 2]
```diff
@@ -40,10 +41,56 @@
...
+                <input id="name" type="text" placeholder="Your Name">
+                <label for="comment-description">Comment:</label>
+                <textarea name="comment" id="comment-description" cols="30" rows="6" placeholder="Comment">
+                </textarea>
+                <button id="submit" type="submit">Comment</button>
+            </form>
+        </div>
+    </div>
+
+    <!-- end  Popup -->
```
Comment (ahzia) [inline src/index.html]: Use Javascript to add this to DOM, It's better our index.html only contains the basic layout which is shared in all parts of the app(header, footer), for sections, popups ... you can use javascript to add it in document load.

**Review by ahzia — CHANGES_REQUESTED** [review]
Hi @benwmx You did a great job, Everything was perfectly done,
I have some suggestions to improve our code.
Suggestion :

- (Required): There should be a way to close the Popup (like when clicking somewhere outside popup or a close button inside popup)

-  (Optional): There is lots of White space in the popup and the width of elements(input boxes) is large, I think if you make it a bit smaller it will look more beautiful.

 

Also Please check the inline comments.
If you have any questions mention me using @ahzia 
Happy coding

**Comment by benwmx** [comment]
Hi, @ahzia I implemented all the changes that you suggested.

**Review by ahzia — APPROVED** [review]
Your changes have been approved, Solve the Conflicts and merge your branch.

**Pull request merged**